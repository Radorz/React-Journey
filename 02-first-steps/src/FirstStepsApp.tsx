import ItemCounter from "./shopping-cart/ItemCounter";



interface ItemInCart {
  productName: string;
  quantity?: number;
}

const itemsInCart: ItemInCart[] = [
  { productName: "Manzana", quantity: 2 },
  { productName: "Pan", quantity: 1 },
  { productName: "Leche", quantity: 3 },
];



export function FirstStepsApp() {

    return (
      <>
        <h1>Hola mundo</h1>
        <p>Esto es un parrafo</p>
        <button>Click me</button>

        <h1>Carrito de compra</h1>

        {itemsInCart.map((item) => (
          <ItemCounter name={item.productName} quantity={item.quantity} />
        ))}
{/* 
        <ItemCounter name="Switch 2" quantity={12} />
        <ItemCounter name="PS5" quantity={2} />
        <ItemCounter name="Sword" quantity={3} /> */}
      </>
    );
}