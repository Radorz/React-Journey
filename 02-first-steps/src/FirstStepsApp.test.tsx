import { afterEach, describe, expect, test, vi } from "vitest";
import { FirstStepsApp } from "./FirstStepsApp";
import { render, screen } from "@testing-library/react";

// vi.mock('./shopping-cart/ItemCounter', () => ({
//     ItemCounter: () => <div data-testid="ItemCounter" />,
// }));


const mockItemCounter = vi.fn( (props: unknown)=>{
    return (
      <div
        data-testid="ItemCounter"        
      />
    );
})

vi.mock("./shopping-cart/ItemCounter", () => ({
default: (props: unknown) => mockItemCounter(props),
}));


// vi.mock("./shopping-cart/ItemCounter", () => ({
//   default: (props: unknown) => <div data-testid="ItemCounter"
//                     name={props.name}
//                     quantity={props.quiantity} />,
// }));

describe("FirstStepsApp", () => {

    afterEach(()=>{
        vi.clearAllMocks()
    })

    test('Should match snpashot', () =>{
        const { container } = render(<FirstStepsApp />);

        expect(container).toMatchSnapshot();

        // screen.debug()
    })


    test('Shoould render the correct number of ItemCounter Components', () => {
        render(<FirstStepsApp />);

        const itemCounters = screen.getAllByTestId('ItemCounter');

        expect(itemCounters.length).toBe(3);

        // screen.debug();
    })

    test('Should render ItemCounter with correct props ', ()=>{
        render(<FirstStepsApp/>)
        
        expect(mockItemCounter).toHaveBeenCalledTimes(3);
        expect(mockItemCounter).toHaveBeenCalledWith({
          name: "Manzana",
          quantity: 2,
        });
        expect(mockItemCounter).toHaveBeenCalledWith({
          name: "Pan",
          quantity: 1,
        });
        expect(mockItemCounter).toHaveBeenCalledWith({
          name: "Leche",
          quantity: 3,
        });
    })

});