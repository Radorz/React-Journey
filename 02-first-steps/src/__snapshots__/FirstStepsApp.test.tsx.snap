// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstStepsApp > Should match snpashot 1`] = `
<div>
  <h1>
    Hola mundo
  </h1>
  <p>
    Esto es un parrafo
  </p>
  <button>
    Click me
  </button>
  <h1>
    Carrito de compra
  </h1>
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
</div>
`;
